src/components/ListItem.jsx

```diff
@@ -1,5 +1,43 @@
+import { useEffect, useState } from 'react';
 import './ListItem.css';
+import { updateItem } from '../api';
 
-export function ListItem({ name }) {
-	return <li className="ListItem">{name}</li>;
+export function ListItem({
+	name,
+	itemId,
+	listPath,
+	purchaseDate,
+	nextPurchaseDate,
+}) {
+	let [isPurchased, setIsPurchased] = useState(false);
+
+	useEffect(() => {
+		const currentDate = new Date();
+		const nextPurchase = nextPurchaseDate.toDate();
+		if (purchaseDate) {
+			if (currentDate < nextPurchase) {
+				setIsPurchased(true);
+			} else {
+				setIsPurchased(false);
+			}
+		}
+	}, []);
```

Comment:
It looks like we’re currently using `nextPurchaseDate` to set the `isPurchased` state, but we should actually be using `dateLastPurchased` for this purpose. The `nextPurchaseDate` isn’t quite right for this logic, as discussed in [this comment](https://github.com/the-collab-lab/tcl-75-smart-shopping-list/pull/39#discussion_r1747834411)!

To align with our acceptance criteria, we need to check if `dateLastPurchased` is within the last 24 hours. You can use the `getFutureDate` function to determine if 24 hours have passed since the purchase. This will ensure that the item stays checked for 24 hours after it’s purchased and then automatically becomes unchecked!

Reply:
I just wanted to circle back and say I was wrong to suggest using the `getFutureDate` function to determine if 24 hours have passed since the purchase. After taking a closer look, I found that this function only offsets the current time by the number of days in the offset!

New suggestion: Let's create a new function that takes a date, adds 24 hours to it, and returns a boolean value indicating whether 24 hours have passed or not! 😄